index.js

```diff
@@ -200,7 +205,46 @@ function close (fastify, done) {
   server.close(done)
 }
 
+class ValidateWebSocket extends WebSocket {
+  constructor (...args) {
+    super(...args)
+    this.afterDuplex = false
+    this.validator = null
+    this.strict = false
+  }
+
+  wrapValidation (handler) {
+    return (message, isBinary) => {
+      if (isBinary) return handler(message, isBinary)
+      try {
+        const parsedInput = JSON.parse(message.toString())
+        if (this.validator(parsedInput)) return handler(parsedInput, false)
+        if (this.strict) {
+          return this.close(1003, 'Unsupported payload')
+        }
+        return this.send(JSON.stringify(this.validator.errors))
+      } catch (e) {
+        if (this.strict) {
+          return this.close(1003, 'Unsupported payload')
+        } else {
+          return this.send('Unsupported payload')
+        }
```

Comment:
Instead of adding a strict flag, make the validation error handling configurable.